### DISCUSSION TITLE
Does it make sense to show the Decrypt permanently dialog on a new unseen file?

### DISCUSSION THREAD

**Discussion by robigan** [tejado/obsidian-gpgCrypt#14] [General]
https://github.com/tejado/obsidian-gpgCrypt/blob/003aef79d3f2f9079a0ca27172e2adaee9763949/src/main.ts#L135-L155
When clicking around my diary vault in Obsidian, clicking on new files that have never been encrypted before (I'm retroactively encrypting my diary) are not in the internal Map that tracks encrypted or unencrypted files, so the "Decrypt permanently" dialog option shows, does this even make sense? Of course, opening that file for the first time adds it to that cache and so it then knows what to do with it. Perhaps it might be better to say that if there's a cache miss, then add that to the cache for tracking with its status based on whether the file is encrypted or not.

**Comment by tejado** [comment]
Hi, yeah, I did not find a good way yet how to handle this. What do you mean by "if there's a cache miss, then add that to the cache for tracking with its status based on whether the file is encrypted or not."? Do you mean, I should read the content of the file during the menu opening to check if it is encrypted or not?
